code/modules/busy_space/air_traffic.dm

```diff
@@ -0,0 +1,128 @@
+//Cactus, Speedbird, Dynasty, oh my
+
+var/datum/lore/atc_controller/atc = new/datum/lore/atc_controller
+
+/datum/lore/atc_controller
+	var/delay_max = 5 MINUTES			//How long between ATC traffic, max.  Default is 5 mins.
+	var/delay_min = 1 MINUTES			//How long between ATC traffic, min.  Default is 1 mins.
+	var/backoff_delay = 5 MINUTES		//How long to back off if we can't talk and want to.  Default is 5 mins.
+	var/next_message					//When the next message should happen in world.time
+	var/force_chatter_type				//Force a specific type of messages
+
+	var/squelched = 0					//If ATC is squelched currently
+
+/datum/lore/atc_controller/New()
+	spawn(30 SECONDS) //Lots of lag at the start of a shift.
+		msg("New shift beginning, resuming traffic control.")
+	next_message = world.time + rand(delay_min, delay_max)
+	process()
+
+/datum/lore/atc_controller/proc/process()
+	if(world.time >= next_message)
+		if(squelched)
+			next_message = world.time + backoff_delay
+		else
+			next_message = world.time + rand(delay_min,delay_max)
+			random_convo()
+
+	spawn(1 MINUTES) //We don't really need high-accuracy here.
+		process()
+
+/datum/lore/atc_controller/proc/msg(var/message,var/sender)
+	ASSERT(message)
+	global_announcer.autosay("[message]", sender ? sender : "[using_map.station_short] Space Control")
```

Comment:
Fullstops are handled in each message and the sender name shouldn't contain one.